client/rpcserver/rpcserver.go

```diff
@@ -268,16 +268,39 @@ func New(cfg *Config) (*RPCServer, error) {
 	return s, nil
 }
 
-// Run starts the web server. Satisfies the dex.Runner interface.
-func (s *RPCServer) Run(ctx context.Context) {
+type RpcConn interface {
+	Connect(ctx context.Context) (error, *sync.WaitGroup)
+}
+
+func NewRpcConn(s *RPCServer) RpcConn {
+	return s
+}
+
+// Start uses ConnectionMaster to start the rpc server with the Connect
+// method using dex.Connector interface
+func (s *RPCServer) Start(ctx context.Context) error {
 	// ctx passed to newMarketSyncer when making new market syncers.
 	s.ctx = ctx
+	rpcConn := NewRpcConn(s)
+	connMaster := dex.NewConnectionMaster(rpcConn)
+	err := connMaster.Connect(s.ctx)
+	// If the initial connection returned an error, shut it down to kill the
+	// auto-reconnect cycle.
+	if err != nil {
+		connMaster.Disconnect()
+		return err
+	}
```

Comment:
I think the intent was to use this in core so you can call connMaster.Disconnect() there when shutdown is received.